cpp/src/io/parquet/experimental/dictionary_page_filter.cu

```diff
@@ -46,11 +48,1538 @@ using parquet::detail::ColumnChunkDesc;
 using parquet::detail::decode_error;
 using parquet::detail::PageInfo;
 
+namespace {
```

Comment:
A general concern I have here is that this seems like a reimplementation of huge parts of the decoding code.  Is there no way to refactor things to just use what's already there?  This is adding a lot of technical debt and duplication.